### PR TITLE
Initial implementation of AsyncGetCallTrace

### PR DESCRIPTION
**Pull request by gacholio**
Currently only supported on Linux x86-64.

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
jenkins compile win,xlinux jdk8

**Comment by gacholio** [comment]
None of our tests exercise this, so a compile on the supported platforms and an unsupported platform should be enough.

File: runtime/j9vm/asgct.cpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		J9StackWalkState walkState;
+		walkState.walkThread = currentThread;
+		walkState.skipCount = 0;
+		walkState.maxFrames = parms->depth;
+		walkState.flags = J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY
+			| J9_STACKWALK_RECORD_BYTECODE_PC_OFFSET | J9_STACKWALK_COUNT_SPECIFIED
+			| J9_STACKWALK_ITERATE_FRAMES;
+		walkState.userData1 = (void*)parms->trace->frames;
+		walkState.frameWalkFunction = asyncFrameIterator;
+		UDATA result = BFUjavaVM->walkStackFrames(currentThread, &walkState);
```
Comment (tajila) [inline runtime/j9vm/asgct.cpp:126]: is it possible for an allocation to occur here? ie in `walkBytecodeFrameSlots`?

not sure we can do that in signal handler code

File: runtime/j9vm/asgct.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+asyncFrameIterator(J9VMThread * currentThread, J9StackWalkState * walkState)
+{
+	ASGCT_CallFrame *frame = (ASGCT_CallFrame*)walkState->userData1;
+	J9Method *method = walkState->method;
+	J9JNIMethodID *methodID = currentThread->javaVM->internalVMFunctions->getJNIMethodID(currentThread, method);
+	Assert_SC_notNull(methodID);
+	frame->method_id = (jmethodID)methodID;
+	J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+	if (J9_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccNative)) {
+		frame->lineno = -3;
```
Comment (tajila) [inline runtime/j9vm/asgct.cpp:74]: for these ones we can define something like `AGCT_LINE_NUMBER_NATIVE_METHOD` and `AGCT_LINE_NUMBER_NOT_AVAILABLE` for readability

File: runtime/j9vm/asgct.cpp (hunk 1 above)
Comment (gacholio) [reply runtime/j9vm/asgct.cpp:126]: We aren't iterating slots, so that code will not be executed.

File: runtime/j9vm/asgct.cpp (hunk 2 above)
Comment (gacholio) [reply runtime/j9vm/asgct.cpp:74]: Done.

**Comment by gacholio** [comment]
jenkins compile win,xlinux jdk8

File: runtime/j9vm/asgct.cpp (hunk 1 above)
Comment (gacholio) [reply runtime/j9vm/asgct.cpp:128]: There is also the artifact search cache, but let's deal with that in a subsequent PR.

**Review by tajila — APPROVED** [review]

**Pull request merged**